lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -317,6 +318,32 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 			}
 			$theme_support_data['settings']['color']['defaultGradients'] = $default_gradients;
 
+			if ( ! isset( $theme_support_data['settings']['typography'] ) ) {
+				$theme_support_data['settings']['typography'] = array();
+			}
+			$default_font_sizes = false;
+			if ( current_theme_supports( 'default-font-sizes' ) ) {
+				$default_font_sizes = true;
+			}
```

Comment:
I don't fully understand this logic.
_Is it not supposed to only be false by default if the theme has added custom font sizes?_
Otherwise, wont this be a huge regression for existing classic themes?

Reply:
I mean that the logic should be turned around.

Reply:
> I don't fully understand this logic.

Yeah, it isn't the most clear thing. I chose to copy the logic directly from the [other presets](https://github.com/WordPress/gutenberg/pull/62252/files#diff-995eb1403df37ca3ca196d6198be899a73dbf5e3aba8fec149c0f1ff1d330935R311-R319) so it would be obvious that it works exactly the same as [other presets](https://github.com/WordPress/gutenberg/pull/62252/files#diff-995eb1403df37ca3ca196d6198be899a73dbf5e3aba8fec149c0f1ff1d330935R301-R309). Refactoring them all seemed out of scope for the deadline.

You could shorten the whole thing to this:

```php
$theme_support_data['settings']['typography']['defaultFontSizes'] =
	current_theme_supports( 'default-font-sizes' ) ||
	! isset( $theme_support_data['settings']['typography']['fontSizes'] );
```

>  Is it not supposed to only be false if the theme has added custom font sizes?

The logic, as written in the code, says it's true when `default-font-sizes` _is_ set _or_ if `editor-font-sizes` is _not_ set. Otherwise, it's false.

So it follows that it is false when `editor-font-sizes` _is_ set _and_ `default-font-sizes` is _not_ set.

```php
$disable_default_font_sizes =
	isset( $theme_support_data['settings']['typography']['fontSizes'] ) &&
	! current_theme_supports( 'default-font-sizes' );

$theme_support_data['settings']['typography']['defaultFontSizes'] =
	! $disable_default_font_sizes;
```

Reply:
But classic themes that are not updated will not have default-font-sizes set.

Reply:
Right, `! isset( $theme_support_data['settings']['typography']['fontSizes'] )` will be `true` triggering the default font sizes to show.